src/appleseed/renderer/modeling/bssrdf/oslbssrdf.cpp

```diff
@@ -240,11 +270,42 @@ namespace
                         modes,
                         s);
 
+                apply_layers_attenuation(*c, i, outgoing_dir, incoming_dir, s);
                 madd(value, s, c->get_closure_weight(i));
             }
         }
 
       private:
+        void apply_layers_attenuation(
+            const CompositeClosure& c,
+            const size_t            closure_index,
+            const Vector3f&         outgoing,
+            const Vector3f&         incoming,
+            Spectrum&               value) const
+        {
+            const std::int8_t* layers = c.get_closure_layers(closure_index);
+
+            for (size_t i = 0; i < CompositeClosure::MaxClosureLayers; ++i)
+            {
+                if (layers[i] < 0)
+                    break;
+
+                const size_t layer = static_cast<size_t>(layers[i]);
+                bsdf_from_closure_id(c.get_closure_type(layer))->attenuate_substrate(
+                    c.get_closure_input_values(layer),
+                    c.get_closure_shading_basis(layer),
+                    outgoing,
+                    incoming,
+                    value);
+            }
+        }
+
+        const BSDF* bsdf_from_closure_id(const ClosureID cid) const
+        {
+            assert(cid >= FirstLayeredClosure);
+            return m_glossy_layer_bsdf.get();
```

Comment:
Out of consistency with `bssrdf_from_closure_id()`, return a const ref instead of a const pointer (or change `bssrdf_from_closure_id()` to return a const pointer).